### ISSUE TITLE
isNaN vs Number.isNaN

### ISSUE THREAD

**Issue by rightaway** [eslint/eslint#14545]
**Please describe what the rule should do:**

`isNaN` and `Number.isNaN` are not the same and have slightly different behavior that could lead to subtle bugs if the wrong one is used. There should be a rule to enforce one or the other to prevent accidental mistakes.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
Number.isNaN()  // gives error is isNaN required
isNaN()  // gives error is Number.isNaN required
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It's part of core javascript and an important function.

**Comment by mdjermanovic** [comment]
Hi @rightaway, thanks for the issue!

Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we will only accept new rules when they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, which isn't the case here so I'm closing this issue.

You can make a custom rule, or enforce this by configuring some of the existing rules like [no-restricted-globals](https://eslint.org/docs/rules/no-restricted-globals) and [no-restricted-properties](https://eslint.org/docs/rules/no-restricted-properties):

* [Demo with `isNaN`](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtZ2xvYmFsczogW1xuICBcImVycm9yXCIsXG4gIHtcbiAgICBuYW1lOiBcImlzTmFOXCIsXG4gICAgbWVzc2FnZTogXCJVc2UgJ051bWJlci5pc05hTigpJyBpbnN0ZWFkLlwiXG4gIH1cbl0qL1xuXG5pc05hTihmb28pOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

* [Demo with `Number.isNaN`](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtcHJvcGVydGllczogW1xuICBcImVycm9yXCIsXG4gIHtcbiAgICBvYmplY3Q6IFwiTnVtYmVyXCIsXG4gICAgcHJvcGVydHk6IFwiaXNOYU5cIixcbiAgICBtZXNzYWdlOiBcIlVzZSAnaXNOYU4oKScgaW5zdGVhZC5cIlxuICB9XG5dKi9cblxuTnVtYmVyLmlzTmFOKGZvbyk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

There is also rule [unicorn/prefer-number-properties](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-number-properties.md) in `eslint-plugin-unicorn`.
